public/app/features/dashboard-scene/panel-edit/PanelDataPane/PanelDataAlertingTab.test.tsx

```diff
@@ -284,18 +307,16 @@ describe('PanelAlertTabContent', () => {
     });
   });
 
+  // after updating to RTKQ, the response is already returning the alerts belonging to the panel
   it('Will render alerts belonging to panel and a button to create alert from panel queries', async () => {
-    mocks.api.fetchRules.mockResolvedValue(rules);
-    mocks.api.fetchRulerRules.mockResolvedValue(rulerRules);
-
     dashboard.panels = [panel];
 
     renderAlertTab(dashboard);
 
     const rows = await ui.row.findAll();
-    expect(rows).toHaveLength(1);
+    expect(rows).toHaveLength(2);
```

Comment:
Why did these indices change, are there more alerts linked to the panel now in the mock response?

Reply:
after updating to RTKQ, the response now includes the alerts associated with the panel, thus, we've shifted the filtering logic to the backend. Consequently, as we've mocked 2 alerts, and we no longer filter the panel and dashboard on the frontend, we render 2 alerts.